Models/Database/DataBaseContext.cs

```diff
@@ -11,6 +13,25 @@ public class DataBaseContext : IdentityDbContext<User>
     {
         public DataBaseContext(DbContextOptions<DataBaseContext> options) : base(options) {}
 
+        public override int SaveChanges()
+        {
+            AddTimestamps();
+            return base.SaveChanges();
+        }
+
+        private void AddTimestamps()
+        {
+            var entities = ChangeTracker.Entries().Where(x => x.Entity is BaseEntity && (x.State == EntityState.Added || x.State == EntityState.Modified));
+            foreach (var entity in entities)
```

Comment:
hmmm is this going to iterate each persisting obj?

Reply:
Yes, but it doesn't represent a significant amount of time when storing into the database